File: 05-04-2024_Strictly_Increasing_Array.java
class Solution {
    public int min_operations(int []nums) {
        int n = nums.length, maxi = 1;
        int dp[] = new int[n];
        
        for(int i = 0; i < n; i++)
            dp[i] = 1;
        
        for(int i = 1; i < n; i++) {
            for(int j = 0; j < i; j++) {
                if(nums[i] > nums[j] && (nums[i] - nums[j] >= (i - j))) {
                    dp[i] = Math.max(1 + dp[j], dp[i]);
                    maxi = Math.max(maxi, dp[i]);
                }
            }
        }
        
        return n - maxi;
    }
}

File: 05-04-2024_Strictly_Increasing_Array.cpp
int min_operations(vector<int>& nums) {
        int n = nums.size(), maxi = 1;
        vector<int> dp(n, 1);
        
        for(int i = 1; i < n; i++) {
            for(int j = 0; j < i; j++) {
                if(nums[i] > nums[j] and (nums[i] - nums[j] >= (i - j))) {
                    dp[i] = max(1 + dp[j], dp[i]);
                    maxi = max(maxi, dp[i]);
                }
            }
        }
        
        return n - maxi;
    }
